util/netcdf_generator.py

```diff
@@ -83,6 +83,37 @@ def _filter_params(self, ds, params):
                 ds = ds.drop(key)
         return ds
 
+    def _setup_coordinate_variables(self, ds):
+        """
+        Find the depth variable in the dataset and set assocciated coordinate attributes (10745)
+        :param ds: the dataset on which to operate
+        :return: the dataset with the coordinate variables and associated attributes setup
+        """
+        # find the depth_variable (10745 AC1)
+        depth_variable = find_depth_variable(ds.data_vars)
+        coordinates_key = 'coordinates'
+        coordinate_variables = "time lat lon"
+        if depth_variable:
+            coordinate_variables += " " + depth_variable
+        for var in ds.data_vars:
```

Comment:
ds.data_vars must be a list

Reply:
do you recommend that i make a code change?

Reply:
if it is possible that ds.data_vars is not a list (or you are not sure), then yes

Reply:
ds.data_vars should always be a list.  in every instance where ds.data_vars is being used throughout the stream_engine code it's being used in a for loop to iterate over a list.